long-running-tasks/src/test/java/io/streamnative/pulsar/recipes/task/Soak.java

```diff
@@ -0,0 +1,398 @@
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.math3.distribution.EnumeratedDistribution;
+import org.apache.commons.math3.distribution.NormalDistribution;
+import org.apache.commons.math3.distribution.RealDistribution;
+import org.apache.commons.math3.util.Pair;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.shade.com.fasterxml.jackson.annotation.JsonIgnore;
+import org.apache.pulsar.shade.com.google.common.collect.Streams;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.testcontainers.containers.PulsarContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * This is a soak test that is intended to be executed indpendently and with user-defined parameters
+ * to simulate their expected workloads.
+ */
+@Disabled
+@Slf4j
+@Testcontainers
+public class Soak {
+  private static final DockerImageName pulsarImage =
+      DockerImageName.parse("apachepulsar/pulsar").withTag("2.10.0");
+
+  @Container
+  private static final PulsarContainer pulsar =
+      new PulsarContainer(pulsarImage) {
+        @Override
+        protected void configure() {
+          super.configure();
+          withStartupTimeout(Duration.ofMinutes(3));
+          withEnv("PULSAR_PREFIX_delayedDeliveryTickTimeMillis", "5");
+        }
+      };
+
+  private final Clock clock = Clock.systemUTC();
+  private final Schema<Task> taskSchema = Schema.JSON(Task.class);
+  private final Schema<TaskMetadata> metadataSchema = Schema.JSON(TaskMetadata.class);
+  private final Schema<Result> resultSchema = Schema.JSON(Result.class);
+
+  private PulsarClient client;
+  private Producer<Task> producer;
+  private Consumer<TaskMetadata> consumer;
+
+  private void createResources(String taskTopic) throws Exception {
+    client = PulsarClient.builder().serviceUrl(pulsar.getPulsarBrokerUrl()).build();
+    producer = client.newProducer(taskSchema).topic(taskTopic).enableBatching(false).create();
+    consumer =
+        client
+            .newConsumer(metadataSchema)
+            .topic(taskTopic + "-state")
+            .subscriptionName(randomUUID().toString())
+            .subscribe();
+  }
+
+  @AfterEach
+  void afterEach() throws Exception {
+    producer.close();
+    consumer.close();
+  }
+
+  @Test
+  @Timeout(value = 4, unit = HOURS)
+  void soak() throws Exception {
+    // Modify these parameters to match your expected workload.
+    var soakTimeout = Duration.ofHours(4);
```

Comment:
What's with the magic numbers? The issue describes these as configurable.

Reply:
I think this is fine for a test